src/dbnode/client/options.go

```diff
@@ -354,7 +357,15 @@ func (o *options) Validate() error {
 func (o *options) SetEncodingM3TSZ() Options {
 	opts := *o
 	opts.readerIteratorAllocate = func(r io.Reader) encoding.ReaderIterator {
-		return m3tsz.NewReaderIterator(r, m3tsz.DefaultIntOptimizationEnabled, encoding.NewOptions())
+		return m3tsz.NewReaderIterator(r, nil, m3tsz.DefaultIntOptimizationEnabled, encoding.NewOptions())
+	}
+	return &opts
+}
+
+func (o *options) SetEncodingProto(schema *desc.MessageDescriptor, encodingOpts encoding.Options) Options {
+	opts := *o
+	opts.readerIteratorAllocate = func(r io.Reader) encoding.ReaderIterator {
```

Comment:
Interesting, for now this makes sense, I assume in future we're going to have to have some sort of per namespace data structure?

Reply:
Yep hopefully @Haijuncao 's code will fix this